src/metabase/pulse/render/body.clj

```diff
@@ -243,19 +243,33 @@
    "#c589b9" "#efce8c" "#b5f95c" "#e35850" "#554dbf" "#bec589" "#8cefc6" "#5cc2f9" "#55e350" "#bf4d4f"
    "#89c3c5" "#be8cef" "#f95cd0" "#50e3ae" "#bf974d" "#899bc5" "#ef8cde" "#f95c67"])
 
+(defn- donut-info
+  "Process rows with a minimum slice threshold. Collapses any segments below the threshold given as a percentage (the
+  value 25 for 25%) into a single category as \"Other\". "
+  [threshold-percentage rows]
+  (let [total                    (reduce + 0 (map second rows))
```

Comment:
Are we 100% sure there can't be any `nil` values for the second column? I guess that probably wouldn't make sense if there were

Reply:
I can make it safer but that's the point of `(common/non-nil-rows x-axis-rowfn y-axis-rowfn rows)`. But you're right it could be safer.

Reply:
sorry I didn't see `non-nil-rows`. Carry on

Reply:
its further up the call tree. so it could easily/accidentally be removed. it makes sense to throw an `(fnil + 0)` in there